typescript/src/chain.ts

```diff
@@ -244,6 +244,26 @@ export interface Bridge {
    * Returns the attached WalletRegistry instance.
    */
   walletRegistry(): Promise<WalletRegistry>
+
+  /**
+   * Builds the redemption data required to request a redemption via
+   * @see TBTCToken#approveAndCall - the built data should be passed as
+   * `extraData` the @see TBTCToken#approveAndCall function.
+   * @param redeemer On-chain identifier of the redeemer.
+   * @param walletPublicKey The Bitcoin public key of the wallet. Must be in the
+   *        compressed form (33 bytes long with 02 or 03 prefix).
+   * @param mainUtxo The main UTXO of the wallet. Must match the main UTXO
+   *        held by the on-chain Bridge contract.
+   * @param redeemerOutputScript The output script that the redeemed funds will
+   *        be locked to. Must be un-prefixed and not prepended with length.
+   * @returns The
+   */
+  buildRedemptionData(
```

Comment:
I am not sure if it should be exported into a public API. The goal of `src/ethereum.ts` is to translate domain-level objects into low-level Ethereum chain primitives and perform a call. With this function we allow those low-level primitives to leak outside.

Instead of having `buildRedemptionData`, I would modify `approveAndCall` interface and replace `extraData: Hex` with domain-level objects, as accepted by `buildRedemptionData`. We could also rename `approveAndCall` to `requestRedemption` and explain in the docs it happens via `approveAndCall`. I think this is fine - although `requestRedemption` does not exist on the Solidity code of `TBTC` token, having it in `TBTCToken` Typescript API is fine given we abstract the complexity and make the API human-readable.

Reply:
Totally agree with @pdyraga!

Reply:
9249ee5